src/mqt/ddsim/sampler.py

```diff
@@ -0,0 +1,144 @@
+
+class DDSIMBackendSampler(BaseSampler):
+    _BACKEND = QasmSimulatorBackend()
+
+    def __init__(
+        self,
+        options: dict | None = None,
+        bound_pass_manager: PassManager | None = None,
+        skip_transpilation: bool = False,
+    ):
+        """Initialize a new DDSIM Sampler
+
+        Args:
+            options: Default options.
+            bound_pass_manager: An optional pass manager to run after
+                parameter binding.
+            skip_transpilation: If this is set to True the internal compilation
+                of the input circuits is skipped and the circuit objects
+                will be directly executed when this objected is called.
+        Raises:
+            ValueError: If backend is not provided
+        """
+
+        super().__init__(options=options)
+        self._transpile_options = Options()
+        self._bound_pass_manager = bound_pass_manager
+        self._preprocessed_circuits: list[QuantumCircuit] | None = None
+        self._transpiled_circuits: list[QuantumCircuit] = []
+        self._skip_transpilation = skip_transpilation
+
+    @property
+    def transpiled_circuits(self) -> list[QuantumCircuit]:
+        if self._skip_transpilation:
+            self._transpiled_circuits = list(self._circuits)
+        elif len(self._transpiled_circuits) < len(self._circuits):
+            # transpile only circuits that are not transpiled yet
+            self._transpile()
+        return self._transpiled_circuits
+
+    @property
+    def backend(self):
+        return self._BACKEND
+
+    @property
+    def transpile_options(self) -> Options:
+        return self._transpile_options
+
+    def set_transpile_options(self, **fields):
+        self._transpile_options.update_options(**fields)
+
+    def _transpile(self):
+        from qiskit import transpile
+
+        start = len(self._transpiled_circuits)
+        self._transpiled_circuits.extend(
+            transpile(
+                self._circuits[start:],
+                target=self.backend.target,
+                **self.transpile_options.__dict__,
+            ),
+        )
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Parameters],
+        **run_options,
+    ) -> PrimitiveJob:
+        for circuit in circuits:
+            self._circuits.append(circuit)
+            self._parameters.append(circuit.parameters)
+
+        job = PrimitiveJob(self._call, parameter_values, **run_options)
+        job.submit()
+        return job
+
+    def _call(
+        self,
+        parameter_values: Sequence[Parameters],
+        **run_options,
+    ) -> SamplerResult:
+        transpiled_circuits = self.transpiled_circuits
+        bound_circuits = self.backend.assign_parameters_from_backend(transpiled_circuits, parameter_values)
+        bound_circuits = self._bound_pass_manager_run(bound_circuits)
+
+        result = [self.backend.run(bound_circuits, **run_options).result()]
```

Comment:
If transpiration to the target gate-set is not being performed automatically anyway, I would expect it to happen here and not in the constructor. 
Compilation can be time-intensive and, thus, should be part of the job and not the construction.

I would even go as far as to say that we could skip transpiration completely and just hope that our Simulator can handle everything that is being thrown at it or the Qiskit circuit import is powerful enough to rewrite things directly if a gate is not supported.

Reply:
I kept that functionality as it was in the original BackendSampler class. However I always wondered why there was no explicit _transpile() method, but instead it was built as an attribute. 
Could you explain me a bit the difference in computational cost between both? I just assumed Qiskit's code was optimized already

Reply:
I might be missing the point here, but
https://qiskit.org/documentation/_modules/qiskit/primitives/backend_sampler.html#BackendSampler
has an explicit `_transpile` method.

Reply:
Yes, I think I didn't phrase the question correctly.
What is the computational advantage of storing information like the circuit list or transpiled circuit list in an instance attribute? Instead of creating a method that directly takes as an argument the circuits that were given as input in the `run method